arangod/Aql/Executor/SingleRemoteModificationExecutor.cpp

```diff
@@ -154,17 +157,21 @@ auto SingleRemoteModificationExecutor<
                                    "missing document reference");
   }
 
-  VPackBuilder inBuilder;
+  velocypack::SupervisedBuffer sbForInBuilder(_info.resourceMonitor());
+  VPackBuilder inBuilder(sbForInBuilder);
   VPackSlice inSlice = VPackSlice::emptyObjectSlice();
   if (_info._input1RegisterId.isValid()) {  // IF NOT REMOVE OR SELECT
     AqlValue const& inDocument = input.getValue(_info._input1RegisterId);
     inBuilder.add(inDocument.slice());
     inSlice = inBuilder.slice();
   }
 
-  std::unique_ptr<VPackBuilder> mergedBuilder = nullptr;
+  velocypack::SupervisedBuffer sbForMergedBuilder(_info.resourceMonitor());
+  auto mergedBuilder = std::make_unique<VPackBuilder>(sbForMergedBuilder);
+  // std::unique_ptr<VPackBuilder> mergedBuilder = nullptr;
```

Comment:
```suggestion
  std::unique_ptr<VPackBuilder> mergedBuilder = nullptr;
```

please go back to this, this is only a placeholder that will eventually be filled only if we enter the following If, which allocates in merge itself.